boards/DCM/Src/App/states/App_FaultState.c

```diff
@@ -14,23 +14,22 @@ static void FaultStateRunOnEntry(struct StateMachine *const state_machine)
 
 static void FaultStateRunOnTick1Hz(struct StateMachine *const state_machine)
 {
-    App_AllStatesRunOnTick1Hz(state_machine);
+    App_SharedStatesRunOnTick1Hz(state_machine);
 }
 
 static void FaultStateRunOnTick100Hz(struct StateMachine *const state_machine)
 {
-    App_AllStatesRunOnTick100Hz(state_machine);
+    App_SharedStatesRunOnTick100Hz(state_machine);
 
     struct DcmWorld *world = App_SharedStateMachine_GetWorld(state_machine);
     struct DcmCanTxInterface *can_tx_interface = App_DcmWorld_GetCanTx(world);
+    struct DcmCanRxInterface *can_rx_interface = App_DcmWorld_GetCanRx(world);
     struct ErrorTable *       error_table = App_DcmWorld_GetErrorTable(world);
-    struct ErrorList          critical_errors;
 
     App_CanTx_SetPeriodicSignal_TORQUE_REQUEST(can_tx_interface, 0.0f);
 
-    App_SharedErrorTable_GetAllCriticalErrors(error_table, &critical_errors);
-
-    if (critical_errors.num_errors == 0)
+    if (App_SharedStates_AreInvertersOK(can_rx_interface) &&
+        App_SharedErrorTable_HasAnyCriticalErrorSet(error_table) == false)
```

Comment:
this might be better
```suggestion
        !App_SharedErrorTable_HasAnyCriticalErrorSet(error_table))
```

Reply:
done